frontend/src/libs/components/meditation-timer/meditation-timer.tsx

```diff
@@ -0,0 +1,69 @@
+import { getValidClassNames } from '#libs/helpers/helpers.js';
+import { useCallback, useState } from '#libs/hooks/hooks.js';
+
+import { DURATION_UNIT, MEDITATION_DURATION } from './constants/constants.js';
+import styles from './styles.module.scss';
+
+type TimerProperties = {
+  onClose: () => void;
+  defaultDuration: string;
+};
+
+const MeditationTimer: React.FC<TimerProperties> = ({
+  onClose,
+  defaultDuration,
+}) => {
+  const [selectedDuration, setSelectedDuration] = useState(defaultDuration);
+
+  const handleClick = useCallback((duration: string) => {
+    return () => {
+      setSelectedDuration(duration);
+    };
+  }, []);
+
+  return (
+    <div className={styles['timer']}>
+      <p className={styles['title']}>Choose Your Duration</p>
+      <div className={styles['duration-container']}>
+        {Object.keys(MEDITATION_DURATION).map((duration) => {
+          return (
+            <button
+              key={duration}
+              className={getValidClassNames(
+                styles['duration-button'],
+                selectedDuration === duration ? styles['active'] : '',
+              )}
+              onClick={handleClick(duration)}
+            >
+              <span
+                className={getValidClassNames(
+                  styles['duration-value'],
+                  selectedDuration === duration ? styles['active-text'] : '',
+                )}
+              >
+                {
+                  MEDITATION_DURATION[
+                    duration as keyof typeof MEDITATION_DURATION
+                  ]
+                }
+              </span>
+              <span
+                className={getValidClassNames(
+                  styles['duration-unit'],
```

Comment:
Looks like you don't use `duration-unit` and `duration-value` styles at all, so you can remove it. I also suggest to make in these span properties styles `color: inherit` and set the color from parent button styles in `duration-button` and `active` classes

Reply:
Agreed with the first point, initially I added those because I though I might need them for styling, turned out not.
Neat idea about inherited color, let me have a look at it.